content/gocourse-2019-05-melbourne.slide

```diff
@@ -815,6 +815,64 @@ Embedding types introduces the possiblity for naming conflicts. Go follows some
 .image _img/crud_puppy.jpg
 .caption J. D. Frazer "Illiad", User Friendly - [[https://en.wikipedia.org/wiki/User_Friendly#Crud_Puppy][Crud Puppy]]
 
+
+* Errors in Go
+
+- `error` is a builtin interface ([[https://golang.org/src/builtin/builtin.go?s=11246:11286#L251][source: error]])
+	type error interface {
+		Error() string
+	}
+- Check errors explicitly
+
+* Errors in Go (continued)
+
+- The [[https://golang.org/pkg/errors/][`errors`]] package has functions for creating new error values
+- Using [[https://golang.org/src/errors/errors.go?s=293:320#L1][`errors.New()`]]:
+
+	func Sqrt(f float64) (float64, error) {
+		if f < 0 {
+			return 0, errors.New("math: square root of negative number")
+		}
+		// implementation
+	}
+- Alternatively using `fmt.Errorf`:
+	if f < 0 {
+		return 0, fmt.Errorf("math: square root of negative number %g", f)
+	}
+
```

Comment:
Nit: double blank line between slides

Reply:
Thanks. Every valid comment is valuable :). I'll fix this in my next PR coz if I push it'll invalidate the approval